src/OpenColorIO/Mutex.h

```diff
@@ -57,30 +57,14 @@ OCIO_NAMESPACE_ENTER
     };
 #endif
 
-    /** Automatically acquire and release lock within enclosing scope. */
-    template <class T>
-    class AutoLock
-    {
-    public:
-        AutoLock() = delete;
-        AutoLock & operator=(const AutoLock &) = delete;
-        AutoLock & operator=(AutoLock &&) = delete;
-
-        AutoLock(T & m) : _m(m) { _m.lock();   }
-        ~AutoLock()             { _m.unlock(); }
-
-    private:
-	   T & _m;
-    };
-
 #ifndef NDEBUG
     // add debug wrappers to mutex
     typedef DebugLock<std::mutex> Mutex;
 #else
     typedef std::mutex Mutex;
 #endif
 
-    typedef AutoLock<std::mutex> AutoMutex;
+    typedef std::lock_guard<std::mutex> AutoMutex;
```

Comment:
A minor change better explaining the design in `debug` mode is to replace `std::mutex` by `Mutex` at line 67.